### PR TITLE
Migrate to using v2 facilities_api in VAOS

### PR DESCRIPTION
**Pull request by JunTaoLuo**
## Are you removing, renaming or moving a folder in this PR?
- [ ] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [x] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).
 - [x] It does not.

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- We've been notified that the v1 APIs are deprecated and we're moving to using v2 facilities_api in VAOS FE.
- Appointments (VAOS) Team
- This is not behind a Flipper toggle

## Related issue(s)

- https://app.zenhub.com/workspaces/appointments-team-603fdef281af6500110a1691/issues/gh/department-of-veterans-affairs/va.gov-team/93949

## Testing done

- Ensure existing unit tests and E2E tests pass

## Screenshots

N/A

## What areas of the site does it impact?

Appointments (VAOS)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/vaos/services/mocks/var/cc_providers.json
Comment (simiadebowale) [inline src/applications/vaos/services/mocks/var/cc_providers.json:1]: @JunTaoLuo - can we move this json file to the `vaos` mocks. generally `var` is considered V0

**Review by simiadebowale — COMMENTED** [review]
Just had a minor comment, otherwise, LGTM!

**Review by vbahinwillit — DISMISSED** [review]
LGTM!

File: src/applications/vaos/services/mocks/var/cc_providers.json
Comment (simiadebowale) [reply src/applications/vaos/services/mocks/var/cc_providers.json:1]: `src/applications/vaos/services/mocks/v2`

Comment (JunTaoLuo) [reply src/applications/vaos/services/mocks/var/cc_providers.json:1]: There are a ton of files under `var/` so maybe we'll do this as a separate task later?

Comment (simiadebowale) [reply src/applications/vaos/services/mocks/var/cc_providers.json:1]: @JunTaoLuo - we can create a cc_provider json in this dir `src/applications/vaos/services/mocks/v2 `. and use it for the mock response. That should be fine for now. We will create a follow up ticket to clean up var Mocks

Comment (JunTaoLuo) [reply src/applications/vaos/services/mocks/var/cc_providers.json:1]: okay done!

**Review by simiadebowale — APPROVED** [review]
LGTM

**Pull request merged**